rosbridge_library/src/rosbridge_library/internal/message_conversion.py

```diff
@@ -198,7 +206,10 @@ def _from_list_inst(inst, rostype):
         return []
 
     # Remove the list indicators from the rostype
-    rostype = re.search(list_tokens, rostype).group(1)
+    try:
+        rostype = re.search(list_tokens, rostype).group(1)
+    except AttributeError:
+        rostype = re.search(bounded_array_tokens, rostype).group(1)
```

Comment:
This was the reason GPSFix messages were failing to parse.

For ROS2 fields, a dynamic array such as `int8[] my_arr` is defined in IDL  as `sequence<int8>`, for which the compiled regex `list_tokens`, defined above, will extract `int8` as the `rostype`.

However, IDL specifies a static array type such as `int8[3] my_arr` with a simple `int8[3]`. In this case, the `list_tokens` regex throws an `AttributeError`, so now I just catch it and try again with a different regex pattern.

See [here](https://design.ros2.org/articles/idl_interface_definition.html) for ROS2 IDL type definitions. There is a table near the bottom that describes array types. This page also gives some insight as to why additional type mappings were required to address past issues (e.g. `boolean` vs `bool`, etc.)